### PR TITLE
Add the `analysis` command group and upload logic

### PR DESCRIPTION
**Pull request by jacoblurye**
Mostly, this PR renames `upload`-related functionality to no longer be `assay`-specific. The key logic change comes with the addition of the `is_analysis` argument to `initiate_upload`.

Wasn't sure of a good way to test this change, but open to suggestions!

Changes accompanying: https://github.com/CIMAC-CIDC/cidc-api-gae/pull/179

### CODE REVIEW

File: cli/api.py [hunk 1]
```diff
@@ -112,8 +124,8 @@ def initiate_assay_upload(assay_name: str, xlsx_file: BinaryIO) -> UploadInfo:
         )
 
 
-def _update_assay_upload_status(job_id: int, etag: str, status: str):
-    """Update the status for an existing assay upload job"""
+def _update_upload_status(job_id: int, etag: str, status: str):
+    """Update the status for an existing upload job"""
     url = _url(f"/assay_uploads/{job_id}")
```
Comment (curlup) [inline cli/api.py]: shouldn't this url also change?

File: cli/cli.py [hunk 2]
```diff
@@ -71,12 +71,43 @@ def upload_assay(assay, xlsx):
...
     Upload data for an assay.
     """
-    upload.upload_assay(assay, xlsx)
+    upload.run_upload(assay, xlsx)
+
+
+#### $ cidc analyses ####
+@click.group()
+@click.option("-i", "--ignore", default=None, hidden=True)
+def analyses(ignore):
```
Comment (curlup) [inline cli/cli.py]: maybe this should be an option of the whole cli thing, not per assay or analysis?

File: cli/api.py (hunk 1 above)
Comment (jacoblurye) [reply cli/api.py]: Nope! Analysis uploads are (perhaps confusingly) stored in the `assay_uploads` table in the database, and this is the default Eve endpoint. Do you think it's important that we store assay and analysis upload jobs separately? Alternatively, should we rename the `assay_uploads` table to be more general?

Comment (curlup) [reply cli/api.py]: I don't have strong opinion here. WDYT

Comment (jacoblurye) [reply cli/api.py]: I'm not sure there's a strong reason to do this in the short term, but this is the sort of detail I could see tripping up a new developer on the project. I'll add a note to the backlog.

**Pull request merged**